packages/python/plotly/plotly/express/_core.py

```diff
@@ -1028,26 +1050,28 @@ def _get_reserved_col_names(args):
                 continue
             elif isinstance(arg, str):  # no need to add ints since kw arg are not ints
                 reserved_names.add(arg)
-            elif isinstance(arg, pd.Series):
-                arg_name = arg.name
-                if arg_name and hasattr(df, arg_name):
-                    in_df = arg is df[arg_name]
+            elif is_into_series(arg):
+                arg_name = nw.from_native(arg, series_only=True).name
+                if arg_name and arg_name in df.columns:
+                    in_df = (
+                        nw.from_native(arg, series_only=True) == df.get_column(arg_name)
+                    ).all()
```

Comment:
we're repeating `nw.from_native(arg, series_only=True)` - granted it should be cheap, but still, could we just compute it once and assign it to a variable?